numpy/_core/src/multiarray/dtypemeta.c

```diff
@@ -1165,17 +1178,21 @@ dtypemeta_wrap_legacy_descriptor(
     dtype_class->scalar_type = descr->typeobj;
     dtype_class->type_num = descr->type_num;
     dt_slots->f = *arr_funcs;
+    dt_slots->setitem = arr_funcs->setitem;
+    dt_slots->getitem = arr_funcs->getitem;
```

Comment:
```suggestion
```
This isn't right, just delete this again, it was a failed try... it's more ocmplicated.